test/browser/browser_asrouter_targeting.js

```diff
@@ -434,3 +435,41 @@ add_task(async function check_pinned_tabs() {
     is(await ASRouterTargeting.Environment.hasPinnedTabs, true, "Should detect pinned tab");
   });
 });
+
+add_task(async function checkCFRPinnedTabsTargetting() {
+  const now = Date.now();
+  const timeMinutesAgo = numMinutes => now - numMinutes * 60 * 1000;
+  const messages = CFRMessageProvider.getMessages();
+  const trigger = {
+    id: "frequentVisits",
+    context: {
+      recentVisits: [
+        {timestamp: timeMinutesAgo(61)},
+        {timestamp: timeMinutesAgo(30)},
+        {timestamp: timeMinutesAgo(1)},
+      ],
+    },
+    param: "github.com",
+  };
+
+  is(await ASRouterTargeting.findMatchingMessage({messages, trigger}), undefined,
+    "should not select PIN_TAB mesage with only 2 visits in past hour");
+
+  trigger.context.recentVisits.push({timestamp: timeMinutesAgo(59)});
+  is(await ASRouterTargeting.findMatchingMessage({messages, trigger}), messages.find(m => m.id === "PIN_TAB"),
+    "should select PIN_TAB mesage");
+
+  await BrowserTestUtils.withNewTab({gBrowser, url: "about:blank"}, async browser => {
+    let tab = gBrowser.getTabForBrowser(browser);
+    gBrowser.pinTab(tab);
```

Comment:
You should probably unpin at the end of the test or do it a `registerCleanupFunction`.